db/migrations/2019_10_15_000000_create_news_table.php

```diff
@@ -0,0 +1,145 @@
+<?php
+declare(strict_types=1);
+
+namespace Engelsystem\Migrations;
+
+use Carbon\Carbon;
+use Engelsystem\Database\Migration\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Capsule\Manager as DB;
+use stdClass;
+
+/**
+ * This migration creates the news table and copies the existing News table records to the new one.
+ */
+class CreateNewsTable extends Migration
+{
+    use ChangesReferences;
+
+    private const DB_DATETIME_FORMAT = 'Y-m-d H:i:s';
+
+    /**
+     * Creates the news table, copies the data and drops the News table.
+     */
+    public function up(): void
+    {
+        $hasPreviousNewsTable = $this->schema->hasTable('News');
+
+        if ($hasPreviousNewsTable) {
+            // rename because some SQL DBMS handle identifiers case insensitive
+            $this->schema->rename('News', 'PreviousNews');
+        }
+
+        $this->createNewNewsTable();
+
+        if ($hasPreviousNewsTable) {
+            $this->copyPreviousToNewNewsTable();
+            $this->changeReferences(
+                'PreviousNews',
+                'ID',
+                'news',
+                'id',
+                'unsignedInteger'
+            );
+            $this->schema->drop('PreviousNews');
+        }
+    }
+
+    private function createNewNewsTable(): void
+    {
+        $this->schema->create('news', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('title', 150);
+            $table->text('text');
+            $table->boolean('is_meeting');
+            $table->unsignedInteger('user_id');
+            $table->foreign('user_id')
+                ->references('id')
+                ->on('users');
+            $table->timestamps();
+        });
+    }
+
+    private function copyPreviousToNewNewsTable(): void
+    {
+        /** @var stdClass[] $previousNewsRecords */
+        $previousNewsRecords = $this->schema
+            ->getConnection()
+            ->table('PreviousNews')
+            ->get();
+
+        foreach ($previousNewsRecords as $previousNews) {
+            $date = Carbon::createFromTimestamp($previousNews->Datum)
+                ->format(self::DB_DATETIME_FORMAT);
+
+            DB::table('news')->insert([
```

Comment:
Better to use the News model here (`$news = new News([data]); $news->save();`)

Reply:
Are you sure? In a migration you only want to do plain database stuff. Why?: Imagine the News models changes that much or we decide to rename or delete it in the future. On setup all migrations run one after another. But because of the missing/modifies News model it crashes.

Reply:
On the other side in that way you can validate the data inside the model, hmm